protocol.html

```diff
@@ -688,6 +688,40 @@ <h2 property="schema:name">Notifications</h2>
               <p>A Solid server MUST conform to the LDN specification by implementing the Receiver parts to receive notifications and make Inbox contents available [<cite><a class="bibref" href="#bib-ldn">LDN</a></cite>].</p>
 
               <p>A Solid client MUST conform to the LDN specification by implementing the Sender or Consumer parts to discover the location of a resource’s Inbox, and to send notifications to an Inbox or to retrieve the contents of an Inbox [<cite><a class="bibref" href="#bib-ldn">LDN</a></cite>].</p>
+
+              <section id="inbox-discovery-client-error" inlist="" rel="schema:hasPart" resource="#inbox-discovery-client-error">
+                <h3 property="schema:name">Inbox Discovery on Client Error</h3>
+                <div datatype="rdf:HTML" property="schema:description">
+                  <p>This section extends the LDN protocol to allow the discovery of the Inbox URL of a target resource on client errors (HTTP responses with <code>4xx</code> status codes) [<cite><a class="bibref" href="#bib-rfc7231">RFC7231</a></cite>][<cite><a class="bibref" href="#bib-ldn">LDN</a></cite>].</p>
+
+                  <p>Servers who want to advertise the Inbox URL in an HTTP response with a <code>4xx</code> status code, <code>MUST</code> do one of the following:</p>
+
+                  <ul>
+                    <li>Include the <code>Link</code> header with a <code>rel</code> value of <code>http://www.w3.org/ns/ldp#inbox</code>.</li>
+                    <li>Return an RDF representation [<cite><a class="bibref" href="#bib-rdf11-concepts">RDF11-CONCEPTS</a></cite>], whose encoded RDF graph contains a relation of type <code>http://www.w3.org/ns/ldp#inbox</code>. The subject of that relation is the target resource and the object is the Inbox.</li>
+                  </ul>
```

Comment:
I still think that "the subject of the relation is the target resource" is confusing, because a relation can be thought of as an arrow, and the target of an arrow is the object of the arrow. So I think one should say "the subject of the relation is the resource which returned the error message" or something along those lines.

Reply:
I like the use of "target resource" as standard terminology across the protocol spec.
Do you think that maybe we should add it to the definitions https://solidproject.org/TR/protocol#definitions to make it explicit?

Reply:
"Target resource" is borrowed from RFC 7231, and I prefer not to add it to Protocol's definitions. The *effective request URI* (RFC 7230) is the reconstructed value of the HTTP request-target. The subject of the relation is same as the *effective request URI*. I've considered using *effective request URI* but figured that "target resource" was more readable - also stays close to LDN use.

I'm open to using *effective request URI* or possibly even *Request-URI* (LDP terminology) but didn't want to associate the subject URI close to the "request". LDN uses *target* resource to refer to the entity with an inbox.

Reply:
The problem is that you can standardize meanings in a spec for that spec, but not across specs. If you have a spec that very clearly has a subject and a target then you can define the meaning of target. But otherwise it is a relational term, and you here have two resources that are targets. The 401 resource which is a target of the initial request and the inbox which is the target of the following request.

Reply:
This is not discussing inbox as the target of the following request.

*Target* is only mentioned in context of an entity with an inbox (equivalent to request-target in context of the subject of the relation) - which is what LDN says.